docs: Describe relations with and between specs (#211)
xmldom/xmldom: readme.md

[!["help wanted" issues](https://img.shields.io/github/issues/xmldom/xmldom/help%20wanted?color=darkgreen&style=flat-square)](https://github.com/xmldom/xmldom/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)
[![Mutation report](https://camo.githubusercontent.com/ee312c4ebce7784ce9f785757eba5d6e33e6d950/68747470733a2f2f696d672e736869656c64732e696f2f656e64706f696e743f7374796c653d666c61742675726c3d687474707325334125324625324662616467652d6170692e737472796b65722d6d757461746f722e696f2532466769746875622e636f6d25324662726f647962697473253246786d6c646f6d2532466d6173746572)](https://dashboard.stryker-mutator.io/reports/github.com/brodybits/xmldom/master)

xmldom is a javascript [ponyfill](https://ponyfill.com/) for the following APIs supported in browsers:
- convert an XML string into a DOM tree (`new DOMParser().parseFromString(xml, mimeType)` => `Document`)
- create, access and modify a DOM tree (`new DOMImplementation().createDocument(...)` => `Document`, )
- serialize a DOM tree back into an XML string (`new XMLSerializer().serializeToString(node)` => `string`)

> If your code always/only needs to work in a browser, you should rather not use xmldom. But you should (potentially always) use it, when your code needs to also work in other runtimes like NodeJS or Rhino.

Reviewer: Just a suggestion:

```suggestion
Note that this `xmldom` library is not required if your code targets a modern browser. But this library is recommended if your code needs to also work in other runtimes like NodeJS or Rhino.
```

> [The implementation is based on several specs](docs/architecture.svg):

Reviewer: I would list the specs here, and *maybe* embed `docs/architecture.svg` in an implementation section near the end. Or should we make a new page that should include implementation details such as embedding `docs/architecture.svg`?

Author: The section directly after this sentence show and describes the relation to the different specs. But maybe this link is rather misleading. Maybe it's good enough for now to not link to the achitecture diagram from the readme for now.

## Specs

> ![Related specifications](docs/specs.svg)

Reviewer: nit: I don't know how accessible embedding this SVG would be to the visually impaired.

My understanding is that the visually impaired are able to read many websites with help from tools such as text-to-speech.

And another understanding is that at least in some parts of North America, free websites need to be accessible since they are considered public resources, and some less-abled people can sue website owners for websites that are not accessible enough.

Author: I would assume that an SVG is more accessible than a PNG file, since it contains the text elements in a semantic manner.
I believe the related PUML file might even be less accessible because of it's weird syntax.
But I also have no clue if there would be a way to list this information in a textual manner including the relationships...

Of course I have no clue about the north american regulations.

Any recommendation?

I think that this is in any way a good starting point since it allows easier access to the related specs, even though not to everyone. Once we discussed which ones we consider relevant for our development we should be able to come up with a shortened textual representation with links to those relevant specs.

Reviewer: > I think that this is in any way a good starting point

👍 agreed on my part

> 

Reviewer: I would recommend removing the extra line:

```suggestion
```

### DOM Parsing and Serialization

> From the [W3C DOM Parsing and Serialization (WD 2016)](https://www.w3.org/TR/2016/WD-DOM-Parsing-20160517/) xmldom provides an implementation for the interfaces

Reviewer: ```suggestion
From the [W3C DOM Parsing and Serialization (WD 2016)](https://www.w3.org/TR/2016/WD-DOM-Parsing-20160517/) `xmldom` provides an implementation for the interfaces:
```

- `DOMParser`
- `XMLSerializer`

> The implementation behaves differently than specified in the spec.

Reviewer: ```suggestion
Note that there are some known deviations between this implementation and the W3 specifications.
```
